safemysql.class.php

```diff
@@ -506,7 +526,19 @@ protected function prepareQuery($args)
 				continue;
 			}
 
-			$value = array_shift($args);
+			if ( !$namedPlaceholders || strlen($part) == 2 )
+			{
+				$value = array_shift($values);
```

Comment:
I think we should use it only for if all placeholders are non-named (!$namedPlaceholders) and use index as placeholder name if we use named placeholders. In this case we can reuse value by numeric placeholder name.
I.e.
```
$row = $db->getRow('SELECT * FROM `table` WHERE `username` = ?s OR `email` = ?s:0', array($_POST['user']))
```